src/domain/products/details/index.js

```diff
@@ -50,9 +50,8 @@ const ProductDetail = ({ id }) => {
     })
   }
 
-  if (didFail) {
-    return <NotFound />
-  }
+  if (didFail) return <NotFound />
+  if (!product) return <Spinner />
```

Comment:
We use `eslint curly` so please 

```suggestion
  if (didFail) {
    return <NotFound />
  }
  if (!product) {
    return <Spinner />
  }
```

Also it seems that `Spinner` is not imported

Reply:
oh I wonder why vscode didn't warn me about that, I'll update it!